sale_timesheet_budget/models/project_project.py

```diff
@@ -0,0 +1,108 @@
+# Copyright 2022-2024 Tecnativa - Víctor Martínez
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import _lt, api, fields, models
+
+
+class ProjectProject(models.Model):
+    _inherit = "project.project"
+
+    budget_ids = fields.One2many(
+        comodel_name="project.project.budget",
+        inverse_name="project_id",
+        string="Budgets",
+        copy=False,
+    )
+    budget_amount = fields.Float(compute="_compute_budget_amount")
+
+    @api.depends("budget_ids")
+    def _compute_budget_amount(self):
+        for item in self:
+            item.budget_amount = sum(item.mapped("budget_ids.amount"))
+
+    def action_profitability_budget_item(self):
+        action = self.env["ir.actions.act_window"]._for_xml_id(
+            "sale_timesheet_budget.action_project_project_budget"
+        )
+        action["domain"] = [("project_id", "=", self.id)]
+        return action
+
+    def _get_profitability_labels(self):
+        res = super()._get_profitability_labels()
+        res["budgets"] = (_lt("Budgets"),)
+        return res
+
+    def _get_profitability_items(self, with_action=True):
+        items = super()._get_profitability_items(with_action)
+        if not self.budget_ids:
+            return items
+        last_sequence = len(items["revenues"]["data"])
+        items["revenues"]["data"].append(
+            {
+                "id": "budgets",
+                "sequence": last_sequence + 1,
+                "invoiced": 0,
+                "to_invoice": self.budget_amount,
+                "action": {
+                    "name": "action_profitability_budget_item",
+                    "type": "object",
+                },
+            }
+        )
+        items["revenues"]["total"]["to_invoice"] += self.budget_amount
+        return items
+
+
+class ProjectProjectBudget(models.Model):
+    _name = "project.project.budget"
+    _description = "Project Project Budget"
+    _order = "date, id"
+
+    date = fields.Date()
+    name = fields.Char(string="Concept")
+    project_id = fields.Many2one(
+        comodel_name="project.project", string="Project", required=True
+    )
+    project_sale_order_id = fields.Many2one(
+        compute="_compute_project_sale_order_id",
+        comodel_name="sale.order",
+    )
+    allowed_sale_order_ids = fields.Many2many(
+        comodel_name="sale.order", compute="_compute_allowed_sale_order_ids"
+    )
```

Comment:
Can this field be converted to `binary` instead of `many2many`? It could then be used as a domain similar to [this example](https://github.com/odoo/odoo/blob/7c1abcb8eb3947a9982d071faab57bf6a0137629/addons/account/models/account_tax.py#L1730).